From 1838011672bd0011230db4dc3b4b4c3df1119a53 Mon Sep 17 00:00:00 2001
From: Carlos Munoz <carlos.munoz@caviumnetworks.com>
Date: Wed, 30 Oct 2013 16:16:36 -0700
Subject: [PATCH 304/518] MIPS: Octeon: RapidIO: Set the ipd port, pko port,
 and pko queue base correctly.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 7cd375370b6a14f78b2fc37a4d21d0d07d19d450
Description:

Signed-off-by: Carlos Munoz <carlos.munoz@caviumnetworks.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/net/ethernet/octeon/ethernet.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/octeon/ethernet.c b/drivers/net/ethernet/octeon/ethernet.c
index e693309..76b98fb 100644
--- a/drivers/net/ethernet/octeon/ethernet.c
+++ b/drivers/net/ethernet/octeon/ethernet.c
@@ -930,10 +930,10 @@ static int cvm_oct_probe(struct platform_device *pdev)
 				tx_header.s.lns = 1;
 				tx_header.s.intr = 1;
 				priv->srio_tx_header = tx_header.u64;
-				priv->ipd_port = cvmx_helper_get_ipd_port(interface, mbox >> 1);
+				priv->ipd_port = cvmx_helper_get_ipd_port(interface, mbox);
 				priv->pko_port = priv->ipd_port;
 				priv->key = priv->ipd_port + (0x10000 * mbox);
-				base_queue = cvmx_pko_get_base_queue(priv->ipd_port) + (mbox & 1);
+				base_queue = cvmx_pko_get_base_queue(priv->ipd_port);
 				priv->num_tx_queues = 1;
 				cvm_oct_by_srio_mbox[interface - 4][mbox] = priv;
 			} else {
-- 
1.9.1

